src/effects/backends/effectprocessor.h

```diff
@@ -154,21 +147,23 @@ class EffectProcessorImpl : public EffectProcessor {
             pState = createSpecificState(bufferParameters);
             m_channelStateMatrix[inputHandle][outputHandle] = pState;
         }
-        processChannel(inputHandle, pState, pInput, pOutput, bufferParameters,
-                       enableState, groupFeatures);
+        processChannel(pState, pInput, pOutput,
+                       bufferParameters, enableState, groupFeatures);
     }
 
     void initialize(const QSet<ChannelHandleAndGroup>& activeInputChannels,
-            EffectsManager* pEffectsManager,
+            const QSet<ChannelHandleAndGroup>& registeredOutputChannels,
             const mixxx::EngineParameters& bufferParameters) final {
+        m_registeredOutputChannels = registeredOutputChannels;
+
         for (const ChannelHandleAndGroup& inputChannel : activeInputChannels) {
             if (kEffectDebugOutput) {
                 qDebug() << this << "EffectProcessorImpl::initialize allocating "
                             "EffectStates for input" << inputChannel;
             }
             ChannelHandleMap<EffectSpecificState*> outputChannelMap;
             for (const ChannelHandleAndGroup& outputChannel :
-                    pEffectsManager->registeredOutputChannels()) {
+                    std::as_const(m_registeredOutputChannels)) {
```

Comment:
I think the const& left hand side of the for implies iterating over the set as const.  (QSet as a const_iterator)

Reply:
Nope, the right hand side needs to be const, too. `qAsConst` doesn't work for return values, so we either need to store it in a local const variable or use `std::as_const`.